app/models/check/email.rb

```diff
@@ -0,0 +1,40 @@
+module Check
+  class Email < Base
+    def check
+      if host.present? && port.present?
+        check_smtp(host, port.to_i)
+      else
+        report_failure('Mail server host or port not set')
+      end
+    end
+
+    private
+
+    def check_smtp(host, port)
+      if port_open?(host, port)
+        mark_message('Mail is up and running')
+      else
+        report_failure('Mail is offline')
+      end
+    end
+
+    def host
+      Rails.application.config.action_mailer.smtp_settings[:address]
+    end
+
+    def port
+      Rails.application.config.action_mailer.smtp_settings[:port]
+    end
+
+    def port_open?(ip, port, seconds = 1)
```

Comment:
Keyword arguments. Don't get why we rescue exceptions.

Reply:
Rescues here means what we are unable to connect to port by some reasons - timeout or some network or address errors. We are not interested in the exact reason we are cannot connect, but in the exact fact - this exact port on this address in unreachable.